[CI] Failure in InternalTopHitsTests#testReduceRandom

Built: https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+multijob-unix-compatibility/os=centos/16/console

Reproduces with:
```
./gradlew :server:test \
  -Dtests.seed=6B8570B9A843D345 \
  -Dtests.class=org.elasticsearch.search.aggregations.metrics.InternalTopHitsTests \
  -Dtests.method="testReduceRandom" \
  -Dtests.security.manager=true \
  -Dtests.locale=en-IE \
  -Dtests.timezone=Asia/Bahrain \
  -Dcompiler.java=10 \
  -Druntime.java=8
```

Failure:
```
java.lang.AssertionError: Didn't match expected value:
                          hits: 
                            hits: same [empty list]
                       max_score: expected [null] but was [1.4E-45]
                           total: same [483783]

	at __randomizedtesting.SeedInfo.seed([6B8570B9A843D345:ABB8DCA88E1E4A13]:0)
	at org.elasticsearch.test.ESTestCase.assertEqualsWithErrorMessageFromXContent(ESTestCase.java:1245)
	at org.elasticsearch.search.aggregations.metrics.InternalTopHitsTests.assertReduced(InternalTopHitsTests.java:203)
```

--- a/server/src/test/java/org/elasticsearch/search/aggregations/metrics/InternalTopHitsTests.java
+++ b/server/src/test/java/org/elasticsearch/search/aggregations/metrics/InternalTopHitsTests.java
@@ -79,7 +79,7 @@ public class InternalTopHitsTests extends InternalAggregationTestCase<InternalTo
         int requestedSize = between(1, 40);
         int actualSize = between(0, requestedSize);
 
-        float maxScore = Float.MIN_VALUE;
+        float maxScore = Float.NEGATIVE_INFINITY;
         ScoreDoc[] scoreDocs = new ScoreDoc[actualSize];
         SearchHit[] hits = new SearchHit[actualSize];
         Set<Integer> usedDocIds = new HashSet<>();
@@ -113,7 +113,7 @@ public class InternalTopHitsTests extends InternalAggregationTestCase<InternalTo
             topDocs = new TopDocs(new TotalHits(totalHits, TotalHits.Relation.EQUAL_TO), scoreDocs);
         }
         // Lucene's TopDocs initializes the maxScore to Float.NaN, if there is no maxScore
-        TopDocsAndMaxScore topDocsAndMaxScore = new TopDocsAndMaxScore(topDocs, maxScore == Float.MIN_VALUE ? Float.NaN : maxScore);
+        TopDocsAndMaxScore topDocsAndMaxScore = new TopDocsAndMaxScore(topDocs, maxScore == Float.NEGATIVE_INFINITY ? Float.NaN : maxScore);
 
         return new InternalTopHits(name, from, requestedSize, topDocsAndMaxScore, searchHits, pipelineAggregators, metaData);
     }
@@ -178,7 +178,7 @@ public class InternalTopHitsTests extends InternalAggregationTestCase<InternalTo
     protected void assertReduced(InternalTopHits reduced, List<InternalTopHits> inputs) {
         SearchHits actualHits = reduced.getHits();
         List<Tuple<ScoreDoc, SearchHit>> allHits = new ArrayList<>();
-        float maxScore = Float.MIN_VALUE;
+        float maxScore = Float.NEGATIVE_INFINITY;
         long totalHits = 0;
         for (int input = 0; input < inputs.size(); input++) {
             SearchHits internalHits = inputs.get(input).getHits();
@@ -200,7 +200,7 @@ public class InternalTopHitsTests extends InternalAggregationTestCase<InternalTo
             expectedHitsHits[i] = allHits.get(i).v2();
         }
         // Lucene's TopDocs initializes the maxScore to Float.NaN, if there is no maxScore
-        SearchHits expectedHits = new SearchHits(expectedHitsHits, totalHits, maxScore == Float.MIN_VALUE ? Float.NaN : maxScore);
+        SearchHits expectedHits = new SearchHits(expectedHitsHits, totalHits, maxScore == Float.NEGATIVE_INFINITY ? Float.NaN : maxScore);
         assertEqualsWithErrorMessageFromXContent(expectedHits, actualHits);
     }
 

--- a/server/src/test/java/org/elasticsearch/search/aggregations/metrics/InternalTopHitsTests.java
+++ b/server/src/test/java/org/elasticsearch/search/aggregations/metrics/InternalTopHitsTests.java
@@ -112,7 +112,8 @@ public class InternalTopHitsTests extends InternalAggregationTestCase<InternalTo
         } else {
             topDocs = new TopDocs(new TotalHits(totalHits, TotalHits.Relation.EQUAL_TO), scoreDocs);
         }
-        TopDocsAndMaxScore topDocsAndMaxScore = new TopDocsAndMaxScore(topDocs, maxScore);
+        // Lucene's TopDocs initializes the maxScore to Float.NaN, if there is no maxScore
+        TopDocsAndMaxScore topDocsAndMaxScore = new TopDocsAndMaxScore(topDocs, maxScore == Float.MIN_VALUE ? Float.NaN : maxScore);
 
         return new InternalTopHits(name, from, requestedSize, topDocsAndMaxScore, searchHits, pipelineAggregators, metaData);
     }

